eventlet/zipkin/api.py

```diff
@@ -155,10 +155,11 @@ def build_span(name, trace_id, span_id, parent_id,
 
     @staticmethod
     def build_annotation(value, endpoint=None):
-        if isinstance(value, unicode):
+        if isinstance(value, str):
             value = value.encode('utf-8')
+        assert isinstance(value, bytes)
```

Comment:
please avoid using `assert` in runtime code. Could be replaced by something like:

```python
if not isinstance(value, bytes):
    raise RuntimeError("Only bytes are accepted...")
```

Reply:
Also, why adding this check?
`ttypes.Annotation` expect a string: https://github.com/eventlet/eventlet/blob/master/eventlet/zipkin/_thrift/zipkinCore/ttypes.py#L143

Reply:
That's UTF-8 encoded string, i.e. bytes. AFAICT this code was never ported to Python 3, so when it says 'str()` in the original code it actually meant `bytes`. Notice I'm deleting a usage of `unicode()`... AFAICT from other uses of the code it expects bytes.

This is, to be fair, a guess though. This code is probably completely busted honestly given it wasn't ported to Python 3.

Reply:
ok thanks for your explanation.